src/libsyntax/parse/mod.rs

```diff
@@ -108,6 +108,7 @@ pub fn parse_stream_from_source_str(
         sess.source_map().new_source_file(name, source),
         override_span,
     );
+    *sess.reached_eof.borrow_mut() |= errors.iter().any(|ud| ud.found_delim.is_none());
```

Comment:
`emit_unclosed_delims` doesn't always have access to the `ParseSess`, sometimes it has `ParseSess`, sometimes `Session`, which makes it less clean to do that.

Reply:
`ParseSess` is always reachable from `Session`. You can make `emit_unclosed_delims` take `ParseSess` rather than `Handler` since the latter is reachable from the former.